File: app/src/main/java/com/bis5/fitjourney/adapters/WorkoutHistoryAdapter.java
package com.bis5.fitjourney.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bis5.fitjourney.R;
import com.bis5.fitjourney.models.WorkoutLog;

import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import java.util.Locale;
import java.util.concurrent.TimeUnit;

public class WorkoutHistoryAdapter extends RecyclerView.Adapter<WorkoutHistoryAdapter.WorkoutHistoryViewHolder> {

    public interface OnLogClickListener {
        void onLogClick(WorkoutLog workoutLog);
    }

    private List<WorkoutLog> workoutLogs = new ArrayList<>();
    private final OnLogClickListener onLogClickListener;

    public WorkoutHistoryAdapter(OnLogClickListener onLogClickListener) {
        this.onLogClickListener = onLogClickListener;
    }

    @NonNull
    @Override
    public WorkoutHistoryViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_workout_history, parent, false);
        return new WorkoutHistoryViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull WorkoutHistoryViewHolder holder, int position) {
        WorkoutLog log = workoutLogs.get(position);
        
        SimpleDateFormat sdf = new SimpleDateFormat("MMMM dd, yyyy", Locale.getDefault());
        holder.historyDate.setText(sdf.format(new Date(log.getDateStarted())));

        long finishTime = (log.getDateFinished() != null) ? log.getDateFinished() : log.getDateStarted();
        long durationMillis = finishTime - log.getDateStarted();
        long durationMinutes = TimeUnit.MILLISECONDS.toMinutes(durationMillis);
        holder.historyDuration.setText("Duration: " + durationMinutes + " min");

        holder.itemView.setOnClickListener(v -> onLogClickListener.onLogClick(log));
    }

    @Override
    public int getItemCount() {
        return workoutLogs == null ? 0 : workoutLogs.size();
    }

    public void updateLogs(List<WorkoutLog> newLogs) {
        this.workoutLogs = newLogs;
        notifyDataSetChanged();
    }

    public static class WorkoutHistoryViewHolder extends RecyclerView.ViewHolder {
        final TextView historyDate;
        final TextView historyDuration;

        public WorkoutHistoryViewHolder(@NonNull View itemView) {
            super(itemView);
            historyDate = itemView.findViewById(R.id.tvHistoryDate);
            historyDuration = itemView.findViewById(R.id.tvHistoryDuration);
        }
    }
}

File: app/src/main/java/com/bis5/fitjourney/adapters/WorkoutHistoryAdapter.kt
package com.bis5.fitjourney.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bis5.fitjourney.R
import com.bis5.fitjourney.models.WorkoutLog
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale
import java.util.concurrent.TimeUnit

class WorkoutHistoryAdapter(
    private var workoutLogs: List<WorkoutLog> = emptyList(),
    private val onLogClick: (WorkoutLog) -> Unit
) : RecyclerView.Adapter<WorkoutHistoryAdapter.WorkoutHistoryViewHolder>() {

    class WorkoutHistoryViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val historyDate: TextView = itemView.findViewById(R.id.tvHistoryDate)
        val historyDuration: TextView = itemView.findViewById(R.id.tvHistoryDuration)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): WorkoutHistoryViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_workout_history, parent, false)
        return WorkoutHistoryViewHolder(view)
    }

    override fun onBindViewHolder(holder: WorkoutHistoryViewHolder, position: Int) {
        val log = workoutLogs[position]
        holder.historyDate.text = SimpleDateFormat("MMMM dd, yyyy", Locale.getDefault()).format(Date(log.dateStarted))

        val durationMillis = (log.dateFinished ?: log.dateStarted) - log.dateStarted
        val durationMinutes = TimeUnit.MILLISECONDS.toMinutes(durationMillis)
        holder.historyDuration.text = "Duration: $durationMinutes min"

        holder.itemView.setOnClickListener {
            onLogClick(log)
        }
    }

    override fun getItemCount() = workoutLogs.size

    fun updateLogs(newLogs: List<WorkoutLog>) {
        workoutLogs = newLogs
        notifyDataSetChanged()
    }
}